src/App.tsx

```diff
@@ -15,36 +15,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = 'default',
+  Alphabetical = 'alphabetical',
+  Length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>([...goodsFromServer]); 
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [activeSort, setActiveSort] = useState<SortType>(SortType.Default);
+
+  const resetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setIsReversed(false);
+    setActiveSort(SortType.Default);
+  };
+
+  const sortAlphabetically = () => {
+    setGoods(() => {
+      const sortedGoods = [...goodsFromServer].sort((a, b) =>
+        a.localeCompare(b),
+      );
+
+      return isReversed ? sortedGoods.reverse() : sortedGoods;
+    });
+    setActiveSort(SortType.Alphabetical);
+  };
+
+  const sortByLength = () => {
+    setGoods(() => {
+      const sortedGoods = [...goodsFromServer].sort(
+        (a, b) => a.length - b.length,
+      );
+
+      return isReversed ? sortedGoods.reverse() : sortedGoods;
+    });
+    setActiveSort(SortType.Length);
+  };
+
+  const reverseGoods = () => {
+    setGoods(prevGoods => [...prevGoods].reverse());
+    setIsReversed(prevReversed => !prevReversed);
+  };
+
+  let alphabeticalButtonClass = 'button is-info is-light';
+
+  if (activeSort === SortType.Alphabetical) {
+    alphabeticalButtonClass = 'button is-info';
+  }
+
+  let lengthButtonClass = 'button is-success is-light';
+
+  if (activeSort === SortType.Length) {
+    lengthButtonClass = 'button is-success';
+  }
+
+  let reverseButtonClass = 'button is-warning is-light';
+
+  if (isReversed) {
+    reverseButtonClass = 'button is-warning';
+  }
+
+  const resetButton =
+    goods.join() !== goodsFromServer.join() ? (
```

Comment:
That's a bad idea. What if goods will be an array of objects instead of strings?
You can just check if you have any type of sort or `isReversed` instead